### DISCUSSION TITLE
update Connection refused

### DISCUSSION THREAD

**Discussion by michael1maass** [Kovah/LinkAce#452] [Support]
Hi, 

Can someone decipher the below System Log entry? 

I do work behind a Corporate Firewall. Web calls routinely go throuth a proxy wich I don't know how to configure in Linkace. 
https://updates.linkace.org answers ok on Edge. I am not sure who refuses the connection, linkace org or anything between here and there. 

same phenomenon on checking the Links entered into Linkace. 

Michael 

 https://updates.linkace.org/api/current-version {"userId":1,"exception":"[object] (Illuminate\\Http\\Client\\ConnectionException(code: 0): Connection refused for URI https://updates.linkace.org/api/current-version at C:\\LaufwerkU\\Sites\\linkace\\vendor\\laravel\\framework\\src\\Illuminate\\Http\\Client\\PendingRequest.php:710)

ö

**Comment by michael1maass** [comment]
I should add: Windows 10, IIS 10, Rewrite Module enabled (I suppose)

**Comment by Kovah** [comment]
Oh.. hmm... there actually is no way to specify aa proxy for all outgoing connections to LinkAce. I'll create an issue for this but support for it will most likely not be added in the near future. :/

**Comment by michael1maass** [comment]
Would you support the assumption, that the Proxy is the Problem?

**Reply by Kovah** [reply]
Yes I think so. You mentioned that you are behind a firewall and the "Connection Refused" error sound just like that: the firewall blocking an outgoing request.

**Reply by michael1maass** [reply]
Hi, 
I tried to make the IIS use the system Proxy by  adding 

```
<system.net>
    <defaultProxy>
      <proxy
        proxyaddress="http://10.xxx.xxx.xxx:3128"
        bypassonlocal="true"
      />
    </defaultProxy>
</system.net>
```
to the Web.config in the /public folder. No Luck so far. ;-) 

I suppose I just have to deactivate the Checking option and set it to "good" status... (edited after discovering that I am a Jerk for not paying better attention to your detail) 

One more observation. I can reach https://updates.linkace.org/ with an ok on edge. I can not, however reach the /api through the same way. shouldn't I get an ok or rather a hint that I am missing attributes 

Michael

**Reply by michael1maass** [reply]
one more for the road.... The time in the logfiles ist 2 hrs. off (early) to my system time although I have Set Europe/Berlin as the timezone. 

Michael

**Reply by michael1maass** [reply]
and yes, I set the Date option in PHP.INI _after_ I wrote this and it doesn't seem to fix the phenomenon

**Reply by Kovah** [reply]
I think changing the Proxy to access LinkAce will not work because it's more likely the requests LinkAce _itself_ is sending to the internet, and not you accessing the application. But as I said, there is currently no way to specify a proxy for that. Sorry.

By the way: changing the timezone in your PHP ini does nothing as Linkace handles the timezone on its own. Even if there was an option to change it I would strongly advise you not to do that.

**Comment by michael1maass** [comment]
Copy that, 

Ich vermutete nur, dass das ggf ein Grund für die Ablehnug an der Firewall sein könnte… welche Zeitzone wird denn im Log verwendet GMT? Und warum nicht die eingestellte ZZ in den Settings? 

Kann ich die Linkprüfung generell irgendwo abschalten?

Michael

**Reply by Kovah** [reply]
In den Logs wird immer UTC verwendet.

Die Linkprüfung lässt sich nicht abstellen. Der Fall, dass keinerlei ausgehende Verbindungen möglich ist, ist aktuell einfach nicht vorgesehen. In #453 ist das bereits als Ticket hinterlegt, hat für mich persönlich allerdings keine Priorität, die Funktion müsste von einem Nutzer beigesteuert werden.
